inc/head.php

```diff
@@ -2,18 +2,33 @@
   session_start();
   require_once('servervars.php');
 
+  $kmw_tiers = array('alpha', 'beta', 'stable');
+  $kmw_builds = array();
+
+  if(isset($_REQUEST['version']) && preg_match('/^(\d+)\.(\d+)\.(\d+)$/', $_REQUEST['version'])) {
+    $kmwbuild = $_REQUEST['version'];
+  } else {
+    foreach ($kmw_tiers as $index => $possible_tier) {
+      $kmw_builds[$possible_tier] = get_keymanweb_version($possible_tier);
+    }
+  }
+
   if(isset($_REQUEST['tier']) &&
       in_array($_REQUEST['tier'], array('alpha','beta','stable'), TRUE)) {
     $tier = $_REQUEST['tier'];
   } else {
-    $tier = 'stable';
-  }
+    // Select the 'stable' version unless we're in a beta cycle.
+    $beta_version   = get_major_version($kmw_builds['beta']);
+    $stable_version = get_major_version($kmw_builds['stable']);
 
-  if(isset($_REQUEST['version']) && preg_match('/^(\d+)\.(\d+)\.(\d+)$/', $_REQUEST['version'])) {
-    $kmwbuild = $_REQUEST['version'];
-  } else {
-    $kmwbuild = get_keymanweb_version($tier);
+    if($stable_version < $beta_version) {
```

Comment:
This is not terribly safe because it's a string comparison. Might be better to use [`version_compare()`](https://www.php.net/manual/en/function.version-compare.php) and then you can do it on the full version strings also, no need to extract out the major version.